src/renderer/services/chain/transaction/common.ts

```diff
@@ -58,7 +62,26 @@ export const sendTx$ = ({
       )
 
     case ETHChain:
-      return ETH.sendTx({ asset, recipient, amount, memo, feeOptionKey })
+      if (txType === TxTypes.SWAP) {
+        const assetAddress = isEthAsset(asset)
+          ? ETHAddress
+          : FP.pipe(
+              getEthTokenAddress(asset),
+              O.getOrElse(() => '')
```

Comment:
[suggestion] Extract it into a helper: `getEthAssetAddress: (asset: Asset) => O.Option<Address>` to use same thing at different places